fix: paywall modal no longer blocks disabling premium features

ordehi opened the pull request:
## Summary
Paywall modal was showing when trying to **disable** premium features like white labeling or password protection - now it only shows when enabling them.

Re: https://posthoghelp.zendesk.com/agent/tickets/46091

## Test plan
Manually tested all affected toggles (surveys + sharing modal) with features removed from org.

ordehi committed:
fix(paywall): only show paywall when enabling premium features, not disabling

The guardAvailableFeature() check was triggering on every toggle change,
blocking users from disabling features they had previously enabled (or
that were enabled by default). Now only gates when enabling the feature.

Affected toggles:
- Survey 'Hide PostHog branding' checkbox
- Sharing modal 'Password protect' toggle
- Sharing modal 'Show PostHog branding' toggle

rafaeelaudibert commented:
@ordehi you tagged growth team but that's not us, moved it over to platform features team :)

adboio commented on frontend/src/lib/components/Sharing/SharingModal.tsx:
```diff
@@ -237,11 +237,16 @@ export function SharingModalContent({
                                                 }
-                                                onChange={(passwordRequired: boolean) =>
-                                                    guardAvailableFeature(AvailableFeature.ADVANCED_PERMISSIONS, () =>
+                                                onChange={(passwordRequired: boolean) => {
```
ahh ok i misunderstood the variable - so `passwordRequired` here is a form value, not like a global auth mechanism (that was my assumption :face_palm:)  
  
this makes sense as a safeguard - assuming `guardAvailableFeature` has no business logic, and this is just something we missed before, i.e. the business logic of "only show modal if they tried to enable the feature" should have been here all along

adboio reviewed (approved):
this makes sense and should work :saluting_face::ship:

_but_ it feels a bit wonky that we have to solve it this way. it's odd that we were ripping the modal onChange without any business logic based on the changed value before this. i feels like `guardAvailableFeature` (or something similar) should be the place for business logic so it's not all inlined here. like, if we make a survey via API, does it also return an error if the user isn't allowed to use some feature? if so, that logic needs to be centralized or we'll inevitably run into this again

not something you need to solve for right now, just wanted to note this

ordehi committed:
refactor: use named parameter for whitelabel toggle onChange

Makes the logic clearer by using showBranding directly from the callback
instead of computing !value from the render prop.
